### PR TITLE
Skip empty checkpoints

### PR DESCRIPTION
**Pull request by pakrym**
T1 client can write checkpoints with a null offset and 0 sequenceNumber, they should be skipped.

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/CheckpointStore/BlobsCheckpointStoreTests.cs [hunk 1]
```diff
@@ -621,7 +621,7 @@ public async Task ListCheckpointsUsesSequenceNumberAsTheStartingPositionWhenNoOf
         /// </summary>
         ///
         [Test]
-        public async Task ListCheckpointsUsesSequenceNumberAsTheStartingPositionWhenOffsetIsNullInLegacyCheckpoint()
+        public async Task ListCheckpointSkipsCheckpointsWhenOffsetIsNullInLegacyCheckpoint()
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/CheckpointStore/BlobsCheckpointStoreTests.cs:624]: Is it never possible to have a valid sequence with a null offset?   If so, we may want to have a couple of additional test scenarios.

File: sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobCheckpointStore/BlobsCheckpointStore.cs [hunk 2]
```diff
@@ -598,7 +600,7 @@ async Task wrapper(CancellationToken token)
         ///       "SequenceNumber":960180
         ///   }
         /// /// </remarks>
-        private static void TryReadLegacyCheckpoint(Span<byte> data, out long? offset, out long? sequenceNumber)
+        private static bool TryReadLegacyCheckpoint(Span<byte> data, out long? offset, out long? sequenceNumber)
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobCheckpointStore/BlobsCheckpointStore.cs:603]: Should we allow a null offset if the sequence number is non-zero?

**Review by jsquire — APPROVED** [review]

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/CheckpointStore/BlobsCheckpointStoreTests.cs (hunk 1 above)
Comment (pakrym) [reply sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/CheckpointStore/BlobsCheckpointStoreTests.cs:624]: T1 was skipping all checkpoints with empty/null offset https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Microsoft.Azure.EventHubs.Processor/src/AzureStorageCheckpointLeaseManager.cs#L125

I might need a test for empty.

File: sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobCheckpointStore/BlobsCheckpointStore.cs (hunk 2 above)
Comment (pakrym) [reply sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobCheckpointStore/BlobsCheckpointStore.cs:603]: What do you mean by allow? Use the sequence number if the offset is 0?

Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobCheckpointStore/BlobsCheckpointStore.cs:603]: It looks like we return `false` when an offset is null, regardless of whether there was a sequence number present.   I'm not sure we should consider that a valid case.

Comment (pakrym) [reply sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobCheckpointStore/BlobsCheckpointStore.cs:603]: >It looks like we return false when an offset is null

https://github.com/Azure/azure-sdk-for-net/pull/18059/files#diff-e78cc94b9611df5a843a75e3888d69b99158921820b2129431ebab5c2e307cfaR630

If the value is null or empty we would continue processing and return true if everything else is fine.
